test/pages/my-account.test.js

```diff
@@ -1,69 +1,74 @@
-
 jest.mock('next/config', () => () => ({
   publicRuntimeConfig: {}
 }))
 
-describe('My Account page', () => {
-  test('renders a default message if the customer has no orders', () => {
-    // Act
-    const wrapper = mount(
-      <Provider store={createMockStore()}>
-        <MyAccountPage orders={{ data: [], loading: false }} dispatch={jest.fn()} />
-      </Provider>
-    )
-
-    // Assert
-    expect(wrapper).toIncludeText('No previous orders found.')
-  })
+// Mock out Router usage in getDerivedStateFromProps
+const originalRouter = Router.router
 
-  test('renders a customers previous orders', () => {
-    // Act
-    const wrapper = mount(
-      <Provider store={createMockStore()}>
-        <MyAccountPage orders={orders} dispatch={jest.fn()} />
-      </Provider>
-    )
+beforeAll(() => {
+  Router.router = {
+    asPath: 'example.com'
+  }
+})
 
-    // Assert
-    expect(wrapper).toIncludeText(`Order Number: ${orders.data[0].reference}`)
-    expect(wrapper).toIncludeText(`Order Number: ${orders.data[1].reference}`)
+afterAll(() => {
+  Router.router = originalRouter
+})
+
+describe('componentDidMount', () => {
+  test('redirects to the first default child when no menu is given', () => {
+    const replaceSpy = jest.spyOn(Router, 'replace').mockImplementation(() => {})
+
+    const wrapper = shallow(<MyAccountPage />, { disableLifecycleMethods: true })
+
+    wrapper.instance().componentDidMount()
 
-    expect(wrapper).toContainReact(<OrderLineItems items={orders.data[0].line_items} />)
-    expect(wrapper).toContainReact(<OrderLineItems items={orders.data[1].line_items} />)
+    expect(replaceSpy).toHaveBeenCalledWith('example.com?menu=Details')
+    expect(wrapper.state().currentMenu).toEqual('Details')
 
-    expect(wrapper).toContainReact(<ShippingAddresses addresses={orders.data[0].shipping_addresses} />)
-    expect(wrapper).toContainReact(<ShippingAddresses addresses={orders.data[1].shipping_addresses} />)
+    replaceSpy.mockRestore()
   })
 
-  test('calls getCustomerOrders on componentDidMount', () => {
-    const dispatch = jest.fn()
-    const expectedFunction = getCustomerOrders().toString()
+  test('redirects to the first default child when an invalid menu is given', () => {
+    Router.router = {
+      asPath: 'example.com?menu=invalid'
+    }
 
-    const wrapper = shallow(<MyAccountPage dispatch={dispatch} orders={{ loading: true }} />, { disableLifecycleMethods: true })
+    const replaceSpy = jest.spyOn(Router, 'replace').mockImplementation(() => {})
+
+    const wrapper = shallow(<MyAccountPage />, { disableLifecycleMethods: true })
 
     wrapper.instance().componentDidMount()
 
-    expect(dispatch).toHaveBeenCalled()
+    expect(replaceSpy).toHaveBeenCalledWith('example.com?menu=Details')
+    expect(wrapper.state().currentMenu).toEqual('Details')
+
+    replaceSpy.mockRestore()
+  })
+
+  test("sets the state and doesn't redirect when a valid menu is given", () => {
```

Comment:
using a mix of double and single quotes in this file

Reply:
This is because there is an apostrophe in the string - it is an exception that eslint allows for.

Reply:
Didn't see that. No worries!